casper/src/main/scala/coop/rchain/casper/util/comm/DeployRuntime.scala

```diff
@@ -72,32 +73,62 @@ object DeployRuntime {
       phloLimit: Long,
       phloPrice: Long,
       validAfterBlock: Int,
-      maybePrivateKey: Option[PrivateKey],
+      maybePrivateKeyFile: Option[String],
       file: String
-  ): F[Unit] =
-    gracefulExit(
-      Sync[F].delay(Try(Source.fromFile(file).mkString).toEither).flatMap {
-        case Left(ex) =>
-          Sync[F].delay(Left(Seq(s"Error with given file: \n${ex.getMessage}")))
-        case Right(code) =>
+  ): F[Unit] = {
+    val readInputFiles: F[Either[Throwable, (String, Option[PrivateKey])]] =
+      Sync[F]
+        .delay(
           for {
-            timestamp <- Sync[F].delay(System.currentTimeMillis())
-
-            //TODO: allow user to specify their public key
-            d = DeployData()
-              .withTimestamp(timestamp)
-              .withTerm(code)
-              .withPhloLimit(phloLimit)
-              .withPhloPrice(phloPrice)
-              .withValidAfterBlockNumber(validAfterBlock)
-              .withTimestamp(timestamp)
-
-            signedData = maybePrivateKey.fold(d)(SignDeployment.sign(_, d))
-
-            response <- DeployService[F].deploy(signedData)
-          } yield response.map(r => s"Response: $r")
-      }
+            code <- Try(Source.fromFile(file).mkString).toEither
+            maybePrivateKeyBase16 <- maybePrivateKeyFile
+                                      .traverse(f => Try(Source.fromFile(f).mkString).toEither)
```

Comment:
`Source` needs to be closed (code and key file). Please use `Resources.withResource()` from `coop.rchain.shared`